lxd/patches.go

```diff
@@ -2475,6 +2476,27 @@ func patchStorageApiDirBindMount(name string, d *Daemon) error {
 	return nil
 }
 
+func patchFixUploadedAt(name string, d *Daemon) error {
+	images, err := db.ImagesGet(d.db, false)
+	if err != nil {
+		return err
+	}
+
+	for _, fingerprint := range images {
+		id, image, err := db.ImageGet(d.db, fingerprint, false, true)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(d.db, "UPDATE images SET upload_date=? WHERE id=?", image.UploadedAt, id)
```

Comment:
I'm a concerned about doing database work/fixes/migrations using the patches system, because:

1) It's not transactional, if a single db.XXX api of the one above fails, but the previous ones have succeeded, you are in an undetermined state, in principle. This might not be a practical concern now, but it will be more with clustering.

2) More importantly, these patches are run after all database updates have been applied, so for example the statement "UPDATE images SET upload_date=? WHERE id=?" could fail if at some point (say) we change the schema of the images table or something like that, and the user upgrades from a version that had a schema compatible with the query in the patch to a version with a schema that is incompatible with the query. We could change the statement to match the new schema, but it could be easily missed since we don't cover each and every upgrade path with integration tests.

In this case, I see no reason for not writing a regular database update instead of a patch.

Reply:
A DB schema update cannot be backported to 2.0, patches can, that's why this is a patch.

It's true I should probably have used this within a transaction, that being said this particular patch can be re-executed perfectly safely so it doesn't actually matter in the end.

Reply:
I see. It's something we'll need to fix then, because in a cluster we'll want patches to be something that is applied locally once per-node, while database changes like this should be applied once cluster-wide by the leader using dqlite.

Reply:
@freeekanayaka I think keeping the DB updates for schema changes makes sense, but we may want to expand the patches system to allow for different scopes. So a patch could be marked as either needing to apply to all the nodes individually (when moving stuff on the filesystem or doing other node-local actions) or be run once for the whole cluster (data migration, changing default profiles, ...)